opossum-tool/OpossumUI
# chore: add commit linter

### Summary of changes

- set up commitlint and enforce via commit hook
- update contributing guide

closes issue #2038

### How can the changes be tested

Make some commits and see potential errors in output.

You can also bypass the hooks using `git commit --no-verify` and then run `yarn lint-commits` to get errors about any commits on top of the main branch. This is also what the CI/CD pipeline does for each PR.

![image](https://github.com/opossum-tool/OpossumUI/assets/46091730/da0b815b-535c-40b3-9f6f-b571ae17122c)
